vendor_consignment_stock

lepistone opened the pull request:
Dependencies:
- [x] OCA/stock-logistics-workflow#44
- [x] OCA/sale-workflow#89

lepistone committed:
initial tests for vendor_consignment_stock

One failing: that's more or less what we need to do with that module.
Later we'll test that it creates a PO, and that validation of the PO
with make the move available.

lepistone committed:
more tests, pending

lepistone committed:
standard mto buy test passes

lepistone committed:
field is_vci in the purcahse order

lepistone committed:
create special vci route and rule

In the future, I'll probably create them automatically, like the module
mrp does.

lepistone committed:
stub procurement workflow

lepistone committed:
create a special vci purchase order

lepistone committed:
VCI PO does not generate pickings

pedrobaeza commented on vendor_consignment_stock/README.rst:
```diff
@@ -0,0 +1,8 @@
+Vendor Consignment Stock
+========================
+
+
```
Please explain the purpose of the module and use OCA description template (https://github.com/OCA/maintainers-tools/tree/master/template/module)

pedrobaeza commented:
Please put stock_ first in the name of the module, to indicate the area/app of the module.

lepistone committed:
when po is confirmed, try to assign the delivery

lepistone committed:
case when sale.cusomer == owner

In this case, no PO should be created, and there should be only one
procurement.

lepistone committed:
choose automatically routes if owner != customer

lepistone committed:
increase sequence of buy vci rule

It is now selected manually when generating procurements from sale order
lines. I don't think I want it to show up by itself.

lepistone committed:
automatically configure the routes for warehouses

This is the exact same logic used in the purchase and mrp modules, i.e.
when a flag is selected in the warehouse, routes are automatically
created, and updated on write.

lepistone committed:
bring back a test, remove another

VCI behaviour without a SO is not supported: a picking still tries to
reserve only stock with the specified owner.

lepistone committed:
rename test file

lepistone committed:
refactor tests a bit

lepistone committed:
rename data xml file

lepistone committed:
use VCI by default

lepistone committed:
doc

lepistone committed:
show is_vci in the PO view

lepistone committed:
pull stock-logistics-workflow

lepistone committed:
depend on sale_owner_stock_sourcing

lepistone committed:
pull sale-workflow

lepistone commented:
@pedrobaeza as we discussed somewhere else, this module could as well start with `purchase_` or `sale_` because it deals with all of them. I'll rename it if you insist though.

BTW, we merged the dependencies and we're now green here!

lepistone committed:
do not select Buy VCI in products: it's automatic

lepistone committed:
fix crash if purchase line ID is False

This can happen if, for example, the product has no supplier.

yvaucher commented on vendor_consignment_stock/README.rst:
```diff
@@ -0,0 +1,46 @@
+Vendor Consignment Stock
```
Here is the template:
https://github.com/OCA/maintainers-tools/blob/master/template/module/README.rst

yvaucher commented on vendor_consignment_stock/model/warehouse.py:
```diff
@@ -0,0 +1,126 @@
+    buy_vci_to_resupply = fields.Boolean(
+        'Purchase from VCI to resupply this warehouse',
+        help="This warehouse can contain Vendor-Supplied-Inventory (VCI) that "
+        "have to be bought before being sold.", default=True)
```
Indentation is not clear here. I prefer the string being indented after `=`

yvaucher commented on vendor_consignment_stock/model/warehouse.py:
```diff
@@ -0,0 +1,126 @@
+                'Can\'t find any generic Buy VCI route.'))
+
+        return {
+            'name': self._format_routename(warehouse, _(' Buy VCI')),
```
You can remove that space before `Buy VCI` it is already in `_format_routename`

```
def _format_routename(self, cr, uid, obj, name, context=None):
    return obj.name + ': ' + name
```

yvaucher commented on vendor_consignment_stock/model/warehouse.py:
```diff
@@ -0,0 +1,126 @@
+        all_routes = super(Warehouse, self).get_all_routes_for_wh(warehouse)
+        if (
+            warehouse.buy_vci_to_resupply and
+            warehouse.buy_vci_pull_id and
```
You can remove this line. If `warehouse.buy_vci_pull_id` is False, `warehouse.buy_vci_pull_id.route_id` will be False too.

yvaucher commented on vendor_consignment_stock/model/warehouse.py:
```diff
@@ -0,0 +1,126 @@
+    @api.model
+    def _get_all_products_to_resupply(self, warehouse):
+        res = super(Warehouse, self)._get_all_products_to_resupply(warehouse)
+        if warehouse.buy_vci_pull_id and warehouse.buy_vci_pull_id.route_id:
```
Same as previous if:

If `warehouse.buy_vci_pull_id` is False, `warehouse.buy_vci_pull_id.route_id` will be False too.

yvaucher commented on vendor_consignment_stock/model/warehouse.py:
```diff
@@ -0,0 +1,126 @@
+        if warehouse.buy_vci_pull_id and warehouse.buy_vci_pull_id.route_id:
+            product_model = self.env['product.product']
+            for product_id in res:
+                for route in product_model.browse(product_id).route_ids:
```
If res is a list of product_id wouldn't a single browse be better then loop of product record set?

By the way I would rename res by product_id or let a comment in docstring

yvaucher commented:
See my inline remarks

lepistone committed:
use owner of sale line as supplier

lepistone committed:
mention OCA in the readme, as per the template

lepistone committed:
lint

lepistone committed:
remove extra space

lepistone committed:
remove useless checks: we get False in that case

lepistone committed:
improve naming, browse only once

lepistone commented:
@yvaucher improved as you suggest. Thanks!

yvaucher commented:
Thanks for the changes :+1:

gurneyalex commented:
:+1: once you added the missing  .pot file.

lepistone committed:
trigger creation of procurement rules on install

This works around odoo/odoo#4853.

lepistone committed:
add i18n templates

lepistone commented:
@gurneyalex done
